presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -281,15 +281,51 @@ public HiveTableHandle getTableHandle(ConnectorSession session, SchemaTableName
     @Override
     public Optional<SystemTable> getSystemTable(ConnectorSession session, SchemaTableName tableName)
     {
-        if (isPartitionsSystemTable(tableName)) {
-            return getPartitionsSystemTable(session, tableName);
+        if (partitionsTableHandler.matches(tableName)) {
+            return getPartitionsSystemTable(session, tableName, partitionsTableHandler.getSourceTableName(tableName));
+        }
+        else if (propertiesTableHandler.matches(tableName)) {
```

Comment:
Nit: the `else` is redundant since the `if` block above returns unconditionally

Reply:
done